name: Update Books Data File

on:
  schedule:
    - cron: '0 5 * * *'
  workflow_dispatch:

jobs:
  update-data:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
        with:
          token: ${{ secrets.GITHUB_TOKEN }}

      - name: Setup Python
        uses: actions/setup-python@v4
        with:
          python-version: '3.9'

      - name: Install dependencies
        run: pip install pandas

      - name: Install dependencies
        run: pip install pandas requests

      - name: Run update script
        run: python .github/Books_store_app/script_gen_books_app_data.py

      - name: Commit and push changes
        run: |
          git config --global user.name "GitHub Actions"
          git config --global user.email "actions@github.com"
          git add .github/Books_store_app/books_data_main.csv
          git add .github/Books_store_app/user_profiles.json
          git add .github/Books_store_app/library.json
          git commit -m "Books_data: $(date '+%Y-%m-%d') - База данных обновлена успешно. Новые данные записаны в books_data_main.csv." || exit 0
          git push origin main 
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} 

      - name: Send Telegram Success Notification
        if: success()
        run: python .github/workflows/send_tg_notification.py "$(date '+%Y-%m-%d') обновление базы данных Books App успешно завершено! В файл books_data_main.csv добавлены новые данные."
        env:
          TELEGRAM_BOT_TOKEN: ${{ secrets.TELEGRAM_BOT_TOKEN }}
          TELEGRAM_CHAT_ID: ${{ secrets.TELEGRAM_CHAT_ID }}

      - name: Send Telegram Failure Notification
        if: failure()
        run: python .github/workflows/send_tg_notification.py "Ошибка выполенения script_gen_books_app_data.py."
        env:
          TELEGRAM_BOT_TOKEN: ${{ secrets.TELEGRAM_BOT_TOKEN }}
          TELEGRAM_CHAT_ID: ${{ secrets.TELEGRAM_CHAT_ID }}
